src/main/java/envoy/client/ui/controller/ChatScene.java

```diff
@@ -204,6 +197,34 @@ private void addContactButtonClicked() {
 		sceneContext.<ContactSearchScene>getController().initializeData(sceneContext, localDB);
 	}
 
+	/**
+	 * Checks the text length of the {@code messageTextArea}, adjusts the
+	 * {@code remainingChars} - Label and checks whether to send the message
+	 * automatically.
+	 *
+	 * @param e the keys that have been entered
+	 * @since Envoy Client v0.1-beta
+	 */
+	@FXML
+	private void checkKeyCombination(KeyEvent e) {
+		// checks whether the text is too long
+		messageTextUpdated();
+		// Automatic sending of messages via (ctrl +) enter
+		checkPostConditions(e);
+	}
+
+	/**
+	 * @param e the keys that have been pressed
+	 * @since Envoy Client v0.1-beta
+	 */
+	@FXML
+	private void checkPostConditions(KeyEvent e) {
+		if (!postButton.isDisabled() && settings.isEnterToSend() && e.getCode() == KeyCode.ENTER
+				|| !settings.isEnterToSend() && e.getCode() == KeyCode.ENTER && e.isControlDown())
```

Comment:
Not entirely sure about that one, but I think the second arm of the condition bypasses the `!postButton.isDisabled()` check, making the `Ctrl + Enter` combination work regardless of the post button's state. Consider testing my suggestion before merging it.

```suggestion
		if (!postButton.isDisabled() && (settings.isEnterToSend() && e.getCode() == KeyCode.ENTER
				|| !settings.isEnterToSend() && e.getCode() == KeyCode.ENTER && e.isControlDown()))
```

Reply:
You're right. I just tested it. Thank god did I already implement a fail-safe in the postMessage method.